03_letters/undrewb/main_test.go

```diff
@@ -0,0 +1,141 @@
+	got := strconv.Quote(buf.String())
+
+	if got != want {
+		t.Errorf("actual: %s does not match expected: %s", got, want)
+	}
+}
+
+var sortLettersTestData = []struct {
+	name  string
+	input string
+	want  string
+}{
+	{name: "lab example", input: "aba", want: "a:2,b:1"},
+	{name: "trailing newline", input: "aba\n", want: "\n:1,a:2,b:1"},
+	{name: "duplicate entries", input: "\t\t\\aba\n", want: "\t:2,\n:1,\\:1,a:2,b:1"},
+	{name: "one letter", input: "a", want: "a:1"},
+	{name: "one repeated letter", input: "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb", want: "b:61"},
+	{name: "repeated entry", input: "abababababababab", want: "a:8,b:8"},
+	{name: "empty string", input: "", want: ""},
+	{name: "random example", input: "blahflehblahbleh", want: "a:2,b:3,e:2,f:1,h:4,l:4"},
+	{name: "whitespace", input: "\t\t   \t\t\t \t\n\n", want: "\t:6,\n:2, :4"},
+	{name: "Katakana",
+		input: "  イロハニホヘト チリヌルヲ ワカヨタレソ ツネナラム ツネナラム ウヰノオクヤマ ケフコエテ アサキユメミシ ヱヒモセスン",
+		want: " :10,ア:1,イ:1,ウ:1,エ:1,オ:1,カ:1,キ:1,ク:1,ケ:1,コ:1,サ:1,シ:1,ス:1,セ:1,ソ:1,タ:1,チ:1,ツ:2,テ:1,ト:1,ナ:2," +
+			"ニ:1,ヌ:1,ネ:2,ノ:1,ハ:1,ヒ:1,フ:1,ヘ:1,ホ:1,マ:1,ミ:1,ム:2,メ:1,モ:1,ヤ:1,ユ:1,ヨ:1,ラ:2,リ:1,ル:1,レ:1,ロ:1,ワ:1,ヰ:1," +
+			"ヱ:1,ヲ:1,ン:1"},
+	{name: "Polish",
+		input: "Pchnąć w tę łódź jeża lub ośm skrzyń fig",
+		want: " :8,P:1,a:1,b:1,c:1,d:1,e:1,f:1,g:1,h:1,i:1,j:1,k:1,l:1,m:1,n:1,o:1,r:1,s:1,t:1,u:1,w:1,y:1,z:1,ó:1,ą:1,ć:1," +
+			"ę:1,ł:1,ń:1,ś:1,ź:1,ż:1"},
+	{name: "thai",
+		input: "  ๏ เป็นมนุษย์สุดประเสริฐเลิศคุณค่า กว่าบรรดาฝูงสัตว์เดรัจฉาน " +
+			"จงฝ่าฟันพัฒนาวิชาการ อย่าล้างผลาญฤๅเข่นฆ่าบีฑาใคร " +
+			"ม่ถือโทษโกรธแช่งซัดฮึดฮัดด่า     หัดอภัยเหมือนกีฬาอัชฌาสัย	" +
+			"ปฏิบัติประพฤติกฎกำหนดใจ พูดจาให้จ๊ะๆ จ๋าๆ น่าฟังเอย ฯ",
+		want: "\t:1, :16,ก:6,ข:1,ค:3,ฆ:1,ง:5,จ:6,ฉ:1,ช:3,ซ:1,ฌ:1,ญ:1,ฎ:1,ฏ:1,ฐ:1,ฑ:1,ฒ:1,ณ:1,ด:10,ต:3,ถ:1," +
+			"ท:1,ธ:1,น:9,บ:3,ป:4,ผ:1,ฝ:2,พ:3,ฟ:2,ภ:1,ม:3,ย:5,ร:9,ฤ:2,ล:3,ว:3,ศ:1,ษ:2,ส:4,ห:4,ฬ:1,อ:6,ฮ:2,ฯ:1,ะ:3,ั" +
+			":12,า:19,ำ:1,ิ:6,ี:2,ึ:1,ื:2,ุ:3,ู:2,เ:7,แ:1,โ:2,ใ:3,ๅ:1,ๆ:2,็:1,่:10,้:2,๊:1,๋:1,์:2,๏:1"},
+	{name: "Russian",
+		input: "Съешь же ещё этих мягких французских булок да выпей чаю",
+		want: " :9,С:1,а:3,б:1,в:1,г:1,д:1,е:4,ж:1,з:1,и:3,й:1,к:3,л:1,м:1,н:1,о:1,п:1,р:1," +
+			"с:1,т:1,у:2,ф:1,х:3,ц:1,ч:1,ш:1,щ:1,ъ:1,ы:1,ь:1,э:1,ю:1,я:1,ё:1"},
+	{name: "Emojis",
+		input: "👾 🙇 💁 🙅 🙆 🙋 🙎 🐵 🙈 🙉 🙊 ❤️ 💔 💌 💕 💞 💓 💗 💖 💘 💝 💟 💜 💛 💚 💙",
+		want: " :25,❤:1,️:1,🐵:1,👾:1,💁:1,💌:1,💓:1,💔:1,💕:1,💖:1,💗:1,💘:1,💙:1,💚:1,💛:1,💜:1,💝:1,💞:1,💟:1," +
+			"🙅:1,🙆:1,🙇:1,🙈:1,🙉:1,🙊:1,🙋:1,🙎:1"},
+}
+
+func TestSortLetters(t *testing.T) {
+	for _, tt := range sortLettersTestData {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			got := strings.Join(sortLetters(letters(tt.input)), ",")
+			if got != tt.want {
+				t.Errorf("sortLetters() = \n%v,\nwant \n%v", got, tt.want)
+			}
+		})
+	}
+}
+
+var lettersTestData = []struct {
+	name  string
+	input string
+	want  map[rune]int
+}{
+	{name: "lab example", input: "aba", want: map[rune]int{97: 2, 98: 1}},
```

Comment:
Can you please update the values for the `want` section of each case so you are using the actual char values rather than int values for the runes? Makes the test cases a lot easier to read.

```go
{name: "lab example", input: "aba", want: map[rune]int{'a': 2, 'b': 1}},
```